test/core/security/grpc_tls_certificate_provider_test.cc

```diff
@@ -532,17 +532,173 @@ TEST_F(GrpcTlsCertificateProviderTest,
 TEST_F(GrpcTlsCertificateProviderTest, SuccessfulKeyCertMatch) {
   absl::StatusOr<bool> status =
       PrivateKeyAndCertificateMatch(private_key_2_, cert_chain_2_);
-  EXPECT_TRUE(status.ok());
+  ASSERT_TRUE(status.ok());
+  EXPECT_TRUE(*status);
+}
+
+TEST_F(GrpcTlsCertificateProviderTest, SuccessfulKeyMultipleCertMatch) {
+  absl::StatusOr<bool> status = PrivateKeyAndCertificateMatch(
+      private_key_2_, /*cert_chain=*/ cert_chain_2_ + cert_chain_);
```

Comment:
This would have failed if it were cert_chain_+cert_chain_2_ right?
Then do we need both test cases single cert and multiple cert. The function anyway ignores all certs after leaf certificate. Maybe just the multiple test case is enough.

Reply:
I've combined both single and multiple cert cases into one `SuccessfulKeyCertMatch` test case.

Reply:
No, don't combine them. Just have the multiple cert case.

Reply:
Please don't combine them. Just have the multiple cert test.